Task Liverpool/src/main/java/pages/SearchResultsPage.java

```diff
@@ -0,0 +1,152 @@
+            firstResult = By.cssSelector("li.m-product__card[data-prodid=\"" + productId + "\"]");
+            scrollToElement(rightArrow);
+            click(rightArrow);
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    public int getTotalNumberOfResultsDisplayed() {
+        int totalResults = findElements(searchResultsDisplayed).size();
+
+        while (clickNextPage()){
+            totalResults += findElementsAfterACurrentElementDisappear(searchResultsDisplayed, firstResult).size();
+        }
+        return totalResults;
+    }
+
+    public void clickListViewButton(){
+        click(listViewButton);
+    }
+
+    public void clickSortProductsBy (SortType sortType) throws InterruptedException {
+
+        click(sortByDropDownList);
+        switch (sortType) {
+            case RELEVANCE:
+                click(sortByRelevance);
+                break;
+
+            case LOWPRICE:
+                click(sortByLowPrice);
+                break;
+
+            case HIGHPRICE:
+                click(sortByHighPrice);
+                break;
+
+            case RATING:
+                click(sortByRating);
+                break;
+
+            case VIEWED:
+                click(sortByViewed);
+                break;
+
+            case SOLD:
+                click(sortBySold);
+                break;
+        }
+        Thread.sleep(1000);
+    }
+
+    public void getPricesInt(){
+        pricesList = findElements(productPrice);
+        pricesIntList = new ArrayList<>();
+
+        for (WebElement prices: pricesList) {
+            pricesIntList.add(Integer.parseInt(prices.getText().substring(1).replace(",", "")));
+        }
+    }
+
+    public void addValuesToPricesIntListFromMoreResultsPages(){
+        while (clickNextPage()){
+            pricesList = findElementsAfterACurrentElementDisappear(productPrice, firstResult);
+            for (WebElement prices: pricesList) {
+                pricesIntList.add(Integer.parseInt(prices.getText().substring(1).replace(",", "")));
+            }
+        }
+    }
+
+    public boolean checkPriceIsSorted(){
+        getPricesInt();
+        addValuesToPricesIntListFromMoreResultsPages();
+
+        for (int i = 0; i < pricesIntList.size() - 1; i++){
+            boolean compare = (pricesIntList.get(i) >= pricesIntList.get(i + 1));
+            if (!compare){
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public void clickFilterLowerThan$500() throws InterruptedException {
+        findElement(filterLowerThan$500Button).click();
+        Thread.sleep(1000);
+    }
+
+    public boolean pricesAreLowerThan$500(){
+        getPricesInt();
+
+        for (int priceInt: pricesIntList
+             ) {
+            if (priceInt > 50000){
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public boolean noResultsFound(){
```

Comment:
is this a test as well?

Reply:
Also moved this to HomeTest now. However I had to make the isElementDisplayedWithTryCatch() method public so I can call it from HomeTest.